dtsync/syncer.go

```diff
@@ -84,3 +98,45 @@ func (s *Syncer) Sync(ctx context.Context, nextCid cid.Cid, sel ipld.Node) error
 		return err
 	}
 }
+
+// has determines if a given cid and selector is in the linksystem for a syncer already.
+func (s *Syncer) has(ctx context.Context, nextCid cid.Cid, sel ipld.Node) bool {
+	getMissingLs := cidlink.DefaultLinkSystem()
+	// trusted because it'll be hashed/verified on the way into the link system when fetched.
+	getMissingLs.TrustedStorage = true
+	getMissingLs.StorageReadOpener = func(lc ipld.LinkContext, l ipld.Link) (io.Reader, error) {
+		r, err := s.ls.StorageReadOpener(lc, l)
+		if err == nil {
+			// Found block read opener, so return it.
+			return r, nil
+		}
+
+		return nil, fmt.Errorf("not available locally")
```

Comment:
nit: [Indent error flow](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow)
```suggestion
		if err != nil {
			return nil, fmt.Errorf("not available locally: %w", err)
		}
		// Found block read opener, so return it.
		return r, nil		
```